kong/pdk/log.lua

```diff
@@ -30,14 +31,22 @@ local setmetatable = setmetatable
 local ngx = ngx
 local kong = kong
 local check_phase = phase_checker.check
+local split = utils.split
 
 
 local _PREFIX = "[kong] "
 local _DEFAULT_FORMAT = "%file_src:%line_src %message"
 local _DEFAULT_NAMESPACED_FORMAT = "%file_src:%line_src [%namespace] %message"
-local PHASES_LOG = phase_checker.phases.log
-
-
+local PHASES = phase_checker.phases
+local PHASES_LOG = PHASES.log
+
+local phases_with_ctx =
+    phase_checker.new(PHASES.certificate,
```

Comment:
The `certificate` phase AFAIK will need OpenResty 1.19.x to work. What I mean is that prior that the `ctx` table is there, but it will be lost after the phase is done. But as it probably quite uncommon to have this on `certificate`, I think it is fine as is, and it starts working when we update the OpenResty, see:

https://github.com/Kong/kong/blob/next/kong/init.lua#L650-L651

Reply:
I removed certificate from the list. If we ever make it not lost, we can add it back, but otherwise it might generate confusion (people could try setting stuff on cert phase and then not find it on the log).